packages/cli/src/databases/repositories/projectRelation.repository.ts

```diff
@@ -8,6 +8,18 @@ export class ProjectRelationRepository extends Repository<ProjectRelation> {
 		super(ProjectRelation, dataSource.manager);
 	}
 
+	/**
+	 * Find the roles of a user in an array of projects.
+	 */
+	async findRoles(userId: string, projectIds: string[]) {
+		const rows = await this.find({
+			where: { userId, projectId: In(projectIds) },
+			select: ['role'],
+		});
+
+		return new Set(rows.map((row) => row.role));
```

Comment:
i'd expect this function to return a map of `projectId -> userRole`. Say I'm an admin in 2 projects, viewer in 3 others and a regular member of 5 others, this would return a single Set containing `['admin', 'viewer', 'member']`. I find this confusing.